src/unclogged/core.clj

```diff
@@ -128,5 +134,5 @@
 
   If you already have a manifold stream, see ->syslog!."
   [conn-opts defaults]
-  (let [sink (s/stream)]
+  (let [sink (create-stream (:buffer-size conn-opts))]
```

Comment:
This is a close call for whether it's best-served with a separate function. Was the original intent to get a var you can redef for tests?

```clojure
(s/stream (conn-opts :buffer-size 0))
```

Of course, this has the advantage that the buffer size continues to be implicit and up to manifold, though it's unlikely that'll change :)

Reply:
Yes, initially I had separated that logic into a separate function just to make sure it works as intended. But it ended up not needing unit tests and it's a private function anyway so it's implicitly tested. I don't have a strong opinion either way; I just found this more readable :)

I'm curious why is the 0 needed? Is it a fallback if buffer-size is nil?

Reply:
@irinarenteria The 0 acts as a default when the key is missing. I don't think manifold treats `(stream nil)` the same as `(stream)` (which has the zero default).

Reply:
That 0 is a default value.

```
=> (def x {:foo 1})
#'x
=> (x :foo)
1
=> (x :foo 2)
1
=> (x :foot 2)
2
```

But I don't know enough about `manifold.stream` to know whether `(s/stream 0)` (buffer size 0) is the same as `(s/stream)` (unbuffered).

Reply:
It _seems_ to be the same:

```
=> (s/stream 0)
=> << stream: {:pending-puts 0, :drained? false, :buffer-size 0, :permanent? false, :type "manifold", :sink? true, :closed? false, :pending-takes 0, :buffer-capacity 0, :source? true} >>
```

Reply:
Looking at this again, `(s/stream (conn-opts :buffer-size 0))` doesn't affect readability enough to warrant abstracting this into a separate function. So I'll push that change soon.